great_expectations/core/expectation_suite.py

```diff
@@ -50,6 +51,7 @@ class ExpectationSuite(SerializableDictDot):
     def __init__(
         self,
         expectation_suite_name,
+        data_context=None,
```

Comment:
Why are we defaulting this to None? If DataContext is required for an Expectation Suite (and we are saying it IS required, otherwise suite functionality doesn't work) it should not default to None.

Reply:
@donaldheppner thank you for bringing up this point. I'm finding that there are a number of different paths through the code that result in the creation of an `ExpectationSuite` object (not just through the `create_expectation_suite()` method in `DataContext`) which would each have to be modified to support this requirement. 

- `Profilers` and sub-classes
- Validators

I'll put together a commit in the morning :)